quiche/src/stream.rs

```diff
@@ -1572,7 +1136,7 @@ impl RangeBuf {
     /// Creates a new `RangeBuf` from the given slice.
     pub fn from(buf: &[u8], off: u64, fin: bool) -> RangeBuf {
         RangeBuf {
-            data: Arc::new(Vec::from(buf)),
+            data: buf.to_vec().into(),
```

Comment:
Thoughts on how this alloc/copy might be avoidable? I get we need something to track what hasn't been ACKed by the peer so we can retransmit if lost, but I wonder if there's a way to leave the data inside the SendBuf circular buffer instead (avoiding alloc+copy). I recognize this would complicate and grow the scope of SendBuf.